documentation/src/docs/pravega-concepts.md

```diff
@@ -11,362 +11,255 @@ You may obtain a copy of the License at
-Kafka's producer-side batching) is related to durability.  Events added to a
-Transaction are durable when the Event is ack'd back to the Writer.  However,
-the Events in the Transaction are NOT visible to readers until the Transaction
-is committed by the Writer.  A Transaction is a lot like a Stream; a Transaction
-is associated with multiple Stream Segments.  When an Event is published into a
+the case of an error, the Transaction is aborted and the results disappear.
+
+A key difference between Pravega's Transactions and similar approaches (Kafka's producer-side batching) vary with the feature durability. Events added to a Transaction are durable when the Event is acknowledged back to the Writer. However, the Events in the Transaction are _not_ visible to Readers until the Transaction is committed by the Writer. A Transaction is a similar to a Stream and is  associated with multiple Stream Segments.  When an Event is published into a
 Transaction, the Event itself is appended to a Stream Segment of the
-Transaction.  Say a Stream had 5 Segments, when a Transaction is created on that
-Stream, conceptually that Transaction also has 5 Segments.  When an Event is
-published into the Transaction, it is routed to the same numbered Segment as if
-it were published to the Stream itself (if the Event would have been placed in
-Segment 3 in the "real" Stream, then it will appear in Segment 3 of the
-Transaction).  When the Transaction is committed, each of the Transaction's
-Segments is automatically appended to the corresponding Segment in the real
-Stream.  If the Stream is aborted, the Transaction, all its Segments and all the
-Events published into the Transaction are removed from Pravega.
+Transaction. 
+
+For example, a Stream has five Segments, when a Transaction is created on that
+Stream, conceptually that Transaction also has five Segments. When an Event is
+published into the Transaction, it is routed and assigned to the same numbered Segment similar to Stream (i.e., Event assigned to **Segment 3** in the Stream will be assigned to **Segment 3** in the Transaction). Once the Transaction is committed, is committed, all the Transaction Segments are automatically appended to their corresponding Segment in the Stream. If the Transaction is aborted, the Transaction, all its Segments and all the Events published into the Transaction are removed from Pravega.
 
 ![Transaction](img/trx.commit.new.png) 
 
-Events published into a Transaction are never visible to the Reader until that
-Transaction is committed.
+Events published into a Transaction are visible to the Reader only after the Transaction is committed.
 
-For more details on working with Transactions, see [Working with Pravega:
+For more details on working with Transactions, please see [Working with Pravega:
 Transactions](transactions.md).
 
 ## State Synchronizers
 
-Pravega is a streaming storage primitive; it can also be thought of as a
```

Comment:
Please, rephrase the beginning of this paragraph: `Pravega implements various building blocks to materialize the Stream primitive. One of such building blocks, namely **State Synchonizer**, is aimed at coordinating processes in a distributed computing environment. Internally, the **State Synchronizer** uses a Pravega Stream to...`